From 6c1ebf327a6dbfc480d718e68e4f320da5fe41ef Mon Sep 17 00:00:00 2001
From: Mihai Serban <mihai.serban@nxp.com>
Date: Fri, 7 Jul 2017 16:13:48 +0300
Subject: [PATCH 2090/5242] MLK-15927-2: ARM64: dts: iMX8QM and iMX8QXP use
 specific SAI compatibility string

commit  12c58e7f5972411f90b1277ecc0bda7dbd72de1a from
https://source.codeaurora.org/external/imx/linux-imx.git

For iMX8QM and iMX8QXP the SAI device driver must to add a constraint
on the period size because of EDMA requirements.

Signed-off-by: Mihai Serban <mihai.serban@nxp.com>
Reviewed-by: Shengjiu Wang <shengjiu.wang@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 .../devicetree/bindings/sound/fsl-sai.txt          |    3 ++-
 arch/arm64/boot/dts/freescale/fsl-imx8qm.dtsi      |    4 ++--
 arch/arm64/boot/dts/freescale/fsl-imx8qxp.dtsi     |    4 ++--
 3 files changed, 6 insertions(+), 5 deletions(-)

diff --git a/Documentation/devicetree/bindings/sound/fsl-sai.txt b/Documentation/devicetree/bindings/sound/fsl-sai.txt
index dd9e597..460afec 100644
--- a/Documentation/devicetree/bindings/sound/fsl-sai.txt
+++ b/Documentation/devicetree/bindings/sound/fsl-sai.txt
@@ -8,7 +8,8 @@ codec/DSP interfaces.
 Required properties:
 
   - compatible		: Compatible list, contains "fsl,vf610-sai",
-			  "fsl,imx6sx-sai" or "fsl,imx6ul-sai"
+			  "fsl,imx6sx-sai", "fsl,imx6ul-sai",
+			  "fsl,imx8qm-sai" or "fsl,imx8mq-sai"
 
   - reg			: Offset and length of the register set for the device.
 
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qm.dtsi b/arch/arm64/boot/dts/freescale/fsl-imx8qm.dtsi
index ada2637..a8a8828 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8qm.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qm.dtsi
@@ -2162,7 +2162,7 @@
 	};
 
 	sai1: sai@59050000 {
-		compatible = "fsl,imx6ul-sai", "fsl,imx6sx-sai";
+		compatible = "fsl,imx8qm-sai";
 		reg = <0x0 0x59050000 0x0 0x10000>;
 		interrupts = <GIC_SPI 316 IRQ_TYPE_LEVEL_HIGH>;
 		clocks = <&clk IMX8QM_AUD_SAI_1_IPG>,
@@ -2179,7 +2179,7 @@
 
 
 	sai0: sai@59040000 {
-		compatible = "fsl,imx6ul-sai", "fsl,imx6sx-sai";
+		compatible = "fsl,imx8qm-sai";
 		reg = <0x0 0x59040000 0x0 0x10000>;
 		interrupts = <GIC_SPI 314 IRQ_TYPE_LEVEL_HIGH>;
 		clocks = <&clk IMX8QM_AUD_SAI_0_IPG>,
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qxp.dtsi b/arch/arm64/boot/dts/freescale/fsl-imx8qxp.dtsi
index 2c881db..d81c690 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8qxp.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qxp.dtsi
@@ -1368,7 +1368,7 @@
 	};
 
 	sai0: sai@59040000 {
-		compatible = "fsl,imx6ul-sai", "fsl,imx6sx-sai";
+		compatible = "fsl,imx8qm-sai";
 		reg = <0x0 0x59040000 0x0 0x10000>;
 		interrupts = <GIC_SPI 314 IRQ_TYPE_LEVEL_HIGH>;
 		clocks = <&clk IMX8QXP_AUD_SAI_0_IPG>,
@@ -1384,7 +1384,7 @@
 	};
 
 	sai1: sai@59050000 {
-		compatible = "fsl,imx6ul-sai", "fsl,imx6sx-sai";
+		compatible = "fsl,imx8qm-sai";
 		reg = <0x0 0x59050000 0x0 0x10000>;
 		interrupts = <GIC_SPI 316 IRQ_TYPE_LEVEL_HIGH>;
 		clocks = <&clk IMX8QXP_AUD_SAI_0_IPG>,
-- 
1.7.9.5

